canon/src/repr_hosted.rs

```diff
@@ -90,16 +90,18 @@ where
         // can we inline the value?
         if len <= buffer.as_ref().len() {
             let mut sink = ByteSink::new(buffer.as_mut(), store.clone());
-            t.write(&mut sink)?;
+            // TODO: Handle this as a host error
+            t.write(&mut sink).unwrap();
 
-            Ok(Repr::Inline {
+            Repr::Inline {
                 bytes: buffer,
                 len: len as u8,
                 _marker: PhantomData,
-            })
+            }
         } else {
-            let id = store.put(&t)?;
-            Ok(Repr::Ident(id))
+            // TODO: Handle errors in Host
+            let id = store.put(&t).unwrap();
```

Comment:
Its unclear to me how the errors are gonna be passed to the host if this I/O fails